client/src/constants/backend_constants.js

```diff
@@ -1,11 +1,89 @@
-import deepFreeze from 'deep-freeze';
 import generatedBackendConstants from '@/../generated/backend_constants.json';
+import forms from '@/constants/forms.js';
+import deepFreeze from 'deep-freeze';
 
 /**
  * Note that these are generated by running `npm run setup` first.
  *
  * @typedef HMModel
  */
-const BACKEND_CONSTANTS = deepFreeze(generatedBackendConstants);
+const mapping = {
+  growth: '_GrowthFactor_params',
+  mdef: 'MassDefinition_params',
+  cosmo: 'cosmo_defaults',
+  filter: 'Filter_params',
+  hmf: 'FittingFunction_params',
+  hod: 'HOD_params',
+  bias: 'Bias_params',
+  halo_concentration: 'CMRelation_params',
+  tracer_concentration: 'CMRelation_params',
+  halo_profile: 'Profile_params',
+  tracer_profile: 'Profile_params',
+  halo_exclusion: 'Exclusion_params',
+  transfer: 'TransferComponent_params',
+};
+
+const BACKEND_CONSTANTS = generatedBackendConstants;
+const form_option_defaults = {};
+const default_form_state = {};
+
+/* halo_model is weird */
+form_option_defaults.halo_model = {};
```

Comment:
Because the backend constants file is where behavior of specific forms is determined now, it would probably be good to put a description of how this behavior needs to be accounted for, and/or to store it in `forms.js` because it looks like this is specific to each form. As it stands, the configuration for each form is being spread among two files. There are a couple ways to account for that. Either with comments, or storing it in one place. But either way, a further description will be needed on how to do this with other forms in the future. Same goes for cosmo in this same file.

Reply:
I was actually able to remove the halo_model mention which is nice. Only exception now is cosmo, which I've documented.

I am a little confused by this feedback, though? There's not really any 'behavior' in backend_constants. It really just reformats the constants generated in the startup script to something that corresponds to client side representation of state instead of server side representation of state. That's all.

Also, I'm not sure what you mean by configuration information being spread across multiple files? Backend constants is exporting the exact same information as before (cosmo defaults exception was included in here before, too), it's just in a slightly different format. Can you clarify?

Reply:
You had explained the reasoning in the other comment. This makes sense.